site/docs/concepts/derivations.md

```diff
@@ -7,700 +7,769 @@ import Mermaid from '@theme/Mermaid';
-our update lambda implementation implies that it's always `true`.
-The _before_ value, however,
-tells us whether this was the very first update of this register,
-and by implication was the first transfer for this pair of accounts.
+While they're more verbose, TypeScript derivations do have certain advantages:
 
-<Mermaid chart={`
-  sequenceDiagram
-    autonumber
-    Derivation->>Update λ: update({sender: alice, recipient: bob})?
-    Update λ-->>Derivation: return "true"
-    Derivation->>Registers: lookup(key = [alice, bob])?
-    Registers-->>Derivation: not found, initialize as "false"
-    Derivation-->>Derivation: Register: "false" => "true"
-    Derivation-)Registers: store(key = [alice, bob], value = "true")
-    Derivation->>Publish λ: publish({sender: alice, recipient: bob}, register = "true", previous = "false")?
-    Publish λ-->>Derivation: return {sender: alice, recipient: bob}
-`}/>
+* TypeScript derivations are strongly typed, and those checks often catch meaningful bugs and defects **before** they're deployed.
+  Your derivation modules also play nicely with VSCode and other developer tooling.
+* TypeScript derivations can use third-party libraries, as well as your native code compiled to WASM.
+* TypeScript can be easier when working with nested or complex document structures.
 
-:::info FAQ
-> Why not have one lambda that can return
-> a register update _and_ derived documents?
-
-**Performance.**
-_Update_ and _publish_ are designed to be
-parallelized and pipelined over many source documents simultaneously,
-while still giving the appearance and correctness of lambdas
-are invoked in strict serial order.
-Notice that (1) above doesn't depend on actually knowing the register
-value, which doesn't happen until (4).
-Many calls like (1) can also happen in parallel,
-so long as their applications to the register value (5)
-happen in the correct order.
-In comparison, a single-lambda design
-would require Flow to await each invocation
-before it can begin the next.
-:::
+Reduction annotations also have some benefits over task state (like SQLite tables):
 
-***
+* Task states are managed by Flow. If they grow to be large (say, you have a **lot** of accounts),
+  then your task must be scaled and could require performance tuning.
+  Reduction annotations, on the other hand, require *no* task-managed state and are extremely efficient.
+* Certain aggregations, such as recursive merging of tree-like structures,
+  are much simpler to express through reduction annotations vs implementing yourself.
 
-Register schemas may also have
-[reduction](schemas.md#reductions) annotations,
-and documents returned by update lambdas
-are _reduced_ into the current register value.
+[See "Where to Accumulate?" for more discussion](#where-to-accumulate).
 
-The compliance department reached out again,
-and this time they need you to identify transfers where
-the sender's account had insufficient funds.
+## Connectors
```

Comment:
I think it would be better to call this section "Supported Languages" or something similar. Although from a Flow architecture viewpoint these *are* connectors, from a user viewpoint, I'm used to connectors as how I integrate with my source and destination. And here what I'm wondering about is *what languages can I use to write transformations?*

You could keep the next paragraph as-is (still explain that they are technically connectors and work as such) but then the next sentence could change to "At present, Flow supports transformations in SQL using SQLite, and TypeScript."

Reply:
Done :+1:

Agreed that heading better characterizes what the user is after.
Architecturally they _are_ connectors, and I didn't want to shy away from that, because in the future we may offer packaged-up use cases which are deployed as docker images / configured in the UI in the same way we do with captures and materializations. But we're not there yet, so no reason to sweat it, and even if we were "Supported Languages" is pretty reasonable.